### PR TITLE
166 nayem terraform initialize

### PR DESCRIPTION
**Pull request by lucifer7663**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [ ] New feature
- [ ] Refactor
- [x] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
- implemented terraform to deploy the lambda
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: terraform/ApiGateWay.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#resource "aws_lambda_permission" "api_gw" {
+#  statement_id  = "AllowExecutionFromAPIGateway"
+#  action        = "lambda:InvokeFunction"
+#  function_name = aws_lambda_function.Hello.function_name
+#  principal     = "apigateway.amazonaws.com"
+#  source_arn = "${aws_apigatewayv2_api.lambda.execution_arn}/*/*"
+#}
+
+
+# ApiGateway
```
[the remark this answers is not in the document]
Comment (lucifer7663) [reply terraform/ApiGateWay.tf]: solved in current commit

File: terraform/dynamoDB.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+resource "aws_dynamodb_table" "nayem-dynamodb-table-terraform" {
+  name           = "nayem-dynamodb-table-terraform"
```
[the remark this answers is not in the document]
Comment (lucifer7663) [reply terraform/dynamoDB.tf]: environment varriable is implemented

File: terraform/terraform_main.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+}
+
+provider "aws" {
+  region  = "ap-northeast-1"
```
[the remark this answers is not in the document]
Comment (lucifer7663) [reply terraform/terraform_main.tf]: environment varriable is implemented

File: terraform/terraform_s3.tf [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  lambda_function {
+    events = ["s3:ObjectRemoved:*"]
+    lambda_function_arn = aws_lambda_function.Hello.arn
+  }
+
+}
+
+variable "nayem-bucket-name" {
+  default = "muhib-nayem-craftsmen-1"
```
[the remark this answers is not in the document]
Comment (lucifer7663) [reply terraform/terraform_s3.tf]: environment varriable is implemented

File: terraform/terraform_sqs.tf [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            "aws:SourceArn" : aws_s3_bucket.s3bucket.arn
+          }
+        }
+      }
+    ]
+  })
+}
+
+variable "nayem-sqs-name" {
+  default = "nayem-craftsmen-sqs-terraform"
```
[the remark this answers is not in the document]
Comment (lucifer7663) [reply terraform/terraform_sqs.tf]: environment varriable is implemented

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**